argilla/src/argilla/_helpers/_datasets_server.py

```diff
@@ -0,0 +1,175 @@
+import requests
+import warnings
+from enum import Enum
+from typing import Any, Dict, Optional, TYPE_CHECKING, Union, List
+
+from argilla._exceptions._hub import DatasetsServerException
+
+if TYPE_CHECKING:
+    from argilla import Settings
+
+DATASETS_SERVER_BASE_URL = "https://datasets-server.huggingface.co"
+DATASETS_SERVER_HEADERS = {"Accept": "application/json"}
+DATASETS_SERVER_TIMEOUT = 30
+
+
+class FeatureType(Enum):
+    """Feature types supported by Argilla in the datasets server."""
+
+    TEXT = "text"
+    CHAT = "chat"
+    IMAGE = "image"
+    LABEL = "label"
+    INT = "int"
+    FLOAT = "float"
+
+
+def _get_dataset_features(repo_id: str, config: Optional[str] = None) -> Dict[str, Dict[str, Any]]:
+    """Get the features of a dataset from the datasets server using the repo_id and config.
+    Extract the features from the response and return them as a dictionary.
+
+    Parameters:
+        repo_id (str): The repository ID of the dataset.
+        config (str, optional): The configuration of the dataset. Defaults to None.
+
+    Returns:
+        Dict[str, Dict[str, Any]]: The features of the dataset.
+    """
+
+    url = f"{DATASETS_SERVER_BASE_URL}/info?dataset={repo_id}"
+
+    try:
+        response = requests.get(url, headers=DATASETS_SERVER_HEADERS, timeout=DATASETS_SERVER_TIMEOUT)
+    except requests.exceptions.RequestException as e:
+        raise DatasetsServerException() from e
+
+    try:
+        response.raise_for_status()
+    except requests.exceptions.HTTPError as e:
+        raise DatasetsServerException() from e
+
+    try:
+        response_json = response.json()
+    except Exception as e:
+        raise DatasetsServerException() from e
+
+    try:
+        dataset_info = response_json["dataset_info"]
+    except KeyError as e:
+        raise DatasetsServerException() from e
```

Comment:
I think we can wrap all the caught exceptions in one `except` block:
```suggestion
    try:
        response = requests.get(url, headers=DATASETS_SERVER_HEADERS, timeout=DATASETS_SERVER_TIMEOUT)
        response.raise_for_status()    
        response_json = response.json()
        dataset_info = response_json["dataset_info"]    
    except (requests.exceptions.RequestException, requests.exceptions.HTTPError, KeyError, Exception) as e:
        raise DatasetsServerException() from e
```